src/assets/dialog-lightbox.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+function createDialogs(selector) {
+  if (!selector) {
+    return console.error("Missing selector argument");
+  }
+
+  const buttonTemplate = document.createElement("button");
+  buttonTemplate.classList.add("lightbox-button");
+  buttonTemplate.setAttribute("aria-haspopup", "dialog");
+
+  const dialogTemplate = document.createElement("dialog");
+  dialogTemplate.classList.add("lightbox");
+  dialogTemplate.innerHTML = `
+    <form method="dialog">
+      <button type="submit">
+      <span aria-hidden>×</span>
+      <span class="sr-only">Close dialog</span>
+      </button>
+      <span aria-hidden></span>
+    </form>
+  `;
+
+  function createDialog(img) {
+    const button = buttonTemplate.cloneNode();
+    const dialog = dialogTemplate.cloneNode(true);
+    const form = dialog.querySelector("form");
+    const span = dialog.querySelector("form > span");
+
+    dialog.style.setProperty("width", img.naturalWidth + "px");
```

Comment:
As far as I can tell, this will happen if the image is not yet loaded. You may want to wait for the onload event of the image instead; unless the button isn't clickable until the image is loaded :)

Reply:
Ah yeah. I'm now only getting the naturalWidth when clicking the button. The button is only visible when the image is loaded so that should be late enough without needing to add onload listeners everywhere.